File: electron/src/renderer/shared/dto/meeting.ts

// ============================================
// MEETING TYPES
// Matches backend schemas
// ============================================

export type MeetingType = 'steering' | 'weekly_status' | 'risk_review' | 'workshop' | 'daily';
export type MeetingPhase = 'pre' | 'in' | 'post';
export type ParticipantRole = 'organizer' | 'required' | 'optional' | 'attendee';
export type ResponseStatus = 'accepted' | 'declined' | 'tentative' | 'pending';

export interface Participant {
  user_id: string;
  role: ParticipantRole;
  response_status: ResponseStatus;
  email?: string;
  display_name?: string;
}

export interface Meeting {
  id: string;
  title: string;
  description?: string;
  organizer_id?: string;
  start_time?: string;
  end_time?: string;
  meeting_type: MeetingType;
  phase: MeetingPhase;
  project_id?: string;
  department_id?: string;
  location?: string;
  teams_link?: string;
  recording_url?: string;
  created_at?: string;
  updated_at?: string;
}

export interface MeetingWithParticipants extends Meeting {
  participants: Participant[];
  organizer?: Participant;
}

export interface MeetingListResponse {
  meetings: Meeting[];
  total: number;
}

export interface MeetingCreate {
  title: string;
  description?: string;
  organizer_id?: string;  // Optional until auth is implemented
  start_time?: string;
  end_time?: string;
  meeting_type: MeetingType;
  project_id?: string;
  department_id?: string;
  location?: string;
  teams_link?: string;
  participant_ids?: string[];
}

export interface MeetingUpdate {
  title?: string;
  description?: string;
  start_time?: string;
  end_time?: string;
  meeting_type?: MeetingType;
  phase?: MeetingPhase;
  location?: string;
  teams_link?: string;
  recording_url?: string;
}

export interface MeetingListResponse {
  meetings: Meeting[];
  total: number;
}

export interface MeetingFilters {
  phase?: MeetingPhase;
  meeting_type?: MeetingType;
  project_id?: string;
  skip?: number;
  limit?: number;
}

export interface MeetingNotifyRecipient {
  email: string;
  name?: string;
  role?: string;
}

export interface MeetingNotifyRequest {
  recipients: MeetingNotifyRecipient[];
  include_agenda?: boolean;
  include_documents?: boolean;
  include_notes?: boolean;
  custom_message?: string;
}

// Meeting type labels
export const MEETING_TYPE_LABELS: Record<MeetingType, string> = {
  steering: 'Steering Committee',
  weekly_status: 'Weekly Status',
  risk_review: 'Risk Review',
  workshop: 'Workshop',
  daily: 'Daily Standup',
};

export const MEETING_PHASE_LABELS: Record<MeetingPhase, string> = {
  pre: 'Chuẩn bị',
  in: 'Đang họp',
  post: 'Hoàn thành',
};

File: backend/app/schemas/meeting.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List
from uuid import UUID


class ParticipantBase(BaseModel):
    user_id: str
    role: str = 'attendee'  # organizer / required / optional / attendee
    response_status: str = 'pending'  # accepted / declined / tentative / pending


class ParticipantCreate(ParticipantBase):
    pass


class Participant(ParticipantBase):
    email: Optional[str] = None
    display_name: Optional[str] = None

    class Config:
        from_attributes = True


class MeetingBase(BaseModel):
    title: str
    description: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    meeting_type: str = 'weekly_status'  # steering / weekly_status / risk_review / workshop / daily
    project_id: Optional[str] = None
    department_id: Optional[str] = None
    location: Optional[str] = None
    teams_link: Optional[str] = None


class MeetingCreate(MeetingBase):
    organizer_id: Optional[str] = None
    participant_ids: Optional[List[str]] = []


class MeetingUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    meeting_type: Optional[str] = None
    phase: Optional[str] = None
    location: Optional[str] = None
    teams_link: Optional[str] = None
    recording_url: Optional[str] = None


class Meeting(MeetingBase):
    id: str
    organizer_id: Optional[str] = None
    phase: str = 'pre'  # pre / in / post
    recording_url: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class MeetingWithParticipants(Meeting):
    participants: List[Participant] = []
    organizer: Optional[Participant] = None


class MeetingList(BaseModel):
    meetings: List[Meeting]
    total: int


class MeetingNotifyRecipient(BaseModel):
    email: str
    name: Optional[str] = None
    role: Optional[str] = None


class MeetingNotifyRequest(BaseModel):
    recipients: List[MeetingNotifyRecipient]
    include_agenda: bool = True
    include_documents: bool = True
    include_notes: bool = False
    custom_message: Optional[str] = None
